### PR TITLE
FIX Regression where pages would be indexed under all subsites

### PR DESCRIPTION
**Pull request by dhensby**

### CODE REVIEW

File: code/search/SearchVariantSubsites.php [hunk 1]
```diff
@@ -83,9 +83,16 @@ public function extractManipulationWriteState(&$writes)
                 continue;
             }
 
+            if (isset($write['fields']['SiteTree:SubsiteID'])) {
```
Comment (tractorcow) [inline code/search/SearchVariantSubsites.php]: Can we add `'File:SubsiteID'` as well?

Comment (dhensby) [inline code/search/SearchVariantSubsites.php]: Files in subsite 0 should be global, but I suppose that means they should still only be indexed once.... So... Yes?

Comment (tractorcow) [inline code/search/SearchVariantSubsites.php]: Yes but files in subsiteid = 3 shouldn't be.

If they are global, they must be indexed in EVERY subsite, not only once.

Comment (dhensby) [inline code/search/SearchVariantSubsites.php]: :(

Comment (tractorcow) [inline code/search/SearchVariantSubsites.php]: Unfortunately Solr acts as an extremely de-normalised store; For any filter (e.g. any given subsite, stage, etc combination) a filter will be created; This is that json-looking string in the indexed data. If a particular model is visible within that filter (such as golbal files) then there needs to be a separate record saved for that combination.

**Comment by dhensby** [comment]
OK - added the `File:SubsiteID` check too

**Comment by tractorcow** [comment]
It looks good, but should we have a test too?

**Comment by dhensby** [comment]
I've added tests, but because the travis fix to find best matching tag/branch isn't in the CI suite isn't running

**Comment by tractorcow** [comment]
I've merged the travis fix; Restarting tests. :)

**Comment by tractorcow** [comment]
See https://github.com/silverstripe/silverstripe-travis-support/pull/38#issuecomment-265860021

**Comment by tractorcow** [comment]
Thats some fine work @dhensby :)

**Comment by dhensby** [comment]
oh darn it, this was meant to be in 2.2!

**Pull request merged**